CMakeLists.txt

```diff
@@ -315,6 +317,14 @@ list(APPEND FAILING_TESTS "Unit_funnelshift")
 list(APPEND FAILING_TESTS "Unit_syncthreads_and")
 list(APPEND FAILING_TESTS "Unit_syncthreads_count")
 list(APPEND FAILING_TESTS "Unit_syncthreads_or")
+# The following three cases have the same scenario: Main thread spawn a set of
+# worker threads and then the main thread return from main without waiting
+# the worker threads to finish. The worker threads segfaults while accessing
+# destructed CHIP-SPV. Tests may sometimes pass but the chances are low after
+# lazy module initialization patch.
+list(APPEND FAILING_TESTS "Unit_hipMemcpyWithStream_MultiThread")
+list(APPEND FAILING_TESTS "Unit_hipStreamPerThread_MultiThread")
+list(APPEND FAILING_TESTS "Unit_hipStreamPerThread_DeviceReset_1")
```

Comment:
I just ran `Unit_hipMemcpyWithStream_MultiThread` 500 times on main and it doesn't crash. It crashed on this branch quite quickly so the failure seems to be introduced in this patch. We should resolve before merging.

Reply:
When talking about race conditions, major optimizations in execution speed can cause races to clash faster. Here the speedup is quite drastic so it's likely the case here.

Reply:
Reverted this change.